edx_exams/apps/lti/views.py

```diff
@@ -101,14 +103,98 @@ def acs(request, lti_config_id):
         return Response(status=400)
 
     if action == 'flag':
+        # NOTE: The flag action is not yet supported.
+        # If implemented, have it modify the exam attempt data (or perhaps another model?)
         success_msg = (
-            f'Flagging exam for user with id {anonymous_user_id} '
+            f'NOTE: The flag action is not yet supported. The following is a placeholder message.'
+            f'Flagging exam attempt for user with id {anonymous_user_id} '
             f'with resource id {resource_id} and attempt number {attempt_number} '
             f'for lti config id {lti_config_id}, status {attempt.status}, exam id {attempt.exam.id}, '
             f'and attempt id {attempt.id}.'
         )
         log.info(success_msg)
-    return Response(status=200)
+
+    # NOTE: The code below is for the 'terminate' action, which is the only action we support currently.
+    # This code and its tests will need to be modified if other ACS actions are implemented.
+    elif action == 'terminate':
+        # Upon receiving a terminate request, the attempt referenced should have their status updated
+        # depending on the reason for termination (reason_code) and the incident_severity (scaling from 0.0 to 1.0).
+        # If the severity is greater than 0.25, then the attempt is marked for secondary review.
+
+        # Get the termination paramenters
+        try:
+            reason_code = data['reason_code']
+            incident_time = data['incident_time']
+            severity = data['incident_severity']
+        except KeyError as err:
+            error_msg = f'ERROR: required parameter {err} was not found.'
+            return Response(status=status.HTTP_400_BAD_REQUEST, data=error_msg)
+
+        # Ensure the incident_severity's a string to comply with the LTI specs
+        # See: http://www.imsglobal.org/spec/proctoring/v1p0#h.rsq8h6qxveab
+        if not isinstance(severity, str):
+            error_msg = 'ERROR: incident_severity must be passed to the ACS endpoint as a string per LTI specs.'
+            return Response(status=status.HTTP_400_BAD_REQUEST, data=error_msg)
+
+        severity = Decimal(severity)
+        SEVERITY_THRESHOLD = 0.25
+        reason_code_description = ASSESSMENT_CONTROL_CODES[reason_code]
+        # Regular submission occurred, but the learner did something
+        # that might be worth marking the attempt for review. Mark the attempt
+        # as requiring review based on the severity level (>0.25 -> needs review)
+        if reason_code == '1':
+            if severity > SEVERITY_THRESHOLD:
+                update_attempt_status(attempt.id, ExamAttemptStatus.second_review_required)
+                success_msg = (
```

Comment:
missing a log statement for this message. But, maybe it would be better to just call log.info() once at the end of these cases?

Also seems like most of these log messages have similar content could we maybe just log the repetitive part on it's own once? `Terminating exam attempt for user ... attempt id {attempt.id}.`

I see the whole log string is sent back to the vendor in the response, I don't think we need to send all of that. The LTI docs have what appears to be optional status field to send back so if we send anything it should be a JSON object just so that response type doesn't change if we use that later.